Reflection_Engine/Query/BaseTypes.cs

```diff
@@ -0,0 +1,58 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Engine.Reflection
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Gets all implemented interfaces and the base type (if present) of a given type.")]
+        [Input("type", "Type whose base types will be returned.")]
+        [Output("baseTypes", "All implemented interfaces and the base type (if present) of the input type.")]
+        public static List<Type> BaseTypes(this Type type)
+        {
+            if (type == null)
+                return new List<Type>();
+
+            List<Type> baseTypes = new List<Type>();
+
+            baseTypes.AddRange(type.GetInterfaces());
+
+            Type baseType = type.BaseType;
```

Comment:
SHould probably keep on digging for more basetypes. More than one level of inheritance is currently breaking BHoM compliance, so should not really come up in practice for BHoM types, but might as well. Simple while loop keeping on checking base type of the base type should be enough.